### PR TITLE
Added docker compose service Solr Cloud

### PR DESCRIPTION
**Pull request by mkalkbrenner**

### CODE REVIEW

**Review by rfay — CHANGES_REQUESTED** [review]
Thanks for this! It's not clear to me whether this tutorial is for Drupal specifically? Please add some context about that.

**Comment by mkalkbrenner** [comment]
Yes, the both docker compose files are drupal-specific. Where should it be reflected?
In the file names, in the README? Both?

**Comment by rfay** [comment]
Just the README and the title, thanks!

**Comment by mkalkbrenner** [comment]
I think that I covered everything now.

**Comment by mkalkbrenner** [comment]
Is there anything else required to get this merged?

**Comment by rfay** [comment]
Thanks for asking @mkalkbrenner ! Yes, there are a couple of things you can do to get this merged faster:

1. Get someone else to manually test this and review it. It helps if it's somebody that has been active in ddev previously, but anybody is a step forward! 
2. Manually test and review other [ddev-contrib PRs](https://github.com/drud/ddev-contrib/pulls) yourself. 
3. Add manual testing instructions either to the PR or the content itself. And explain what cloud mode is, or link to it. (Maybe you did that, but I missed it in a quick skim just now)

I've been digging through a backlog of PRs and issues in ddev and do hope to get to the ddev-contrib PRs before long. We really need an additional maintainer(s). All it really takes is quality-checking the PRs and manually testing them. Unfortunately that can require some mental energy because they often come from a place a person is not already familiar with, like this one with "cloud mode".

**Comment by rfay** [comment]
I guess I thought you were asking how to get it merged faster, and you were really asking if there was anything to be done to bring it to completion. I apologize for misreading your question.

I guess just explaining how to manually test this would be great. It looked pretty ominous to me.

File: docker-compose-services/drupal-search_api_solr-cloud/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+Although ddev has [documented generic Solr support](https://ddev.readthedocs.io/en/stable/users/extend/additional-services/#apache-solr) it is as simple as possible, and supports only a single core named "dev".
+
+Search API Solr 4.1.x provides a ready to use jump-start config-sets for Solr ion stand-alone and in Cloud mode.
+But only running Solr in Cloud mode gives you all features. 
+
+1. Add the search_api_solr module to your project: `ddev composer require drupal/search_api_solr`
+2. Copy [docker-compose.drupal-search_api_solr-cloud.yaml](docker-compose.drupal-search_api_solr-cloud.yaml) to your project's .ddev folder.
+3. `ddev restart` will bring up the new solr container.
+4. Enable the Search API Solr module: `ddev drush en search_api_solr`
+5. Import the config `web/modules/contrib/search_api_solr/jump-start/drupal_configs/search_api.server.solr.yml` in Drupal or create your own search server using the Solr Cloud Connector and the default collection name `drupal`.
```
Comment (rfay) [inline docker-compose-services/drupal-search_api_solr-cloud/README.md:12]: I was following along here and it wasn't clear how to do this step. I went to admin/config/search/search-api/add-server but it wasn't clear there how one would "import config"

**Review by rfay — CHANGES_REQUESTED** [review]
Some minor requests, thanks for this. I got stuck at step 4 as mentioned, Thanks for this recipe!

File: docker-compose-services/drupal-search_api_solr-cloud/README.md (hunk 1 above)
Comment (mkalkbrenner) [reply docker-compose-services/drupal-search_api_solr-cloud/README.md:12]: This is standard drupal. You could either use the UI for configuration import or `drush cim`.

Comment (rfay) [reply docker-compose-services/drupal-search_api_solr-cloud/README.md:12]: Right, I'm asking you to explain that in the text, not to answer it here in the comments. Please push another commit that expands on this. Remember that not everybody who uses your instructions will have the same background you do.

"or create your own search server using the Solr Cloud Connector"  also seemed pretty vague.

File: docker-compose-services/drupal-search_api_solr-cloud/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+6. Edit the Search API Server and replace `localhost` by `[DDEV_SITENAME].ddev.site`
```
[the remark this answers is not in the document]
Comment (mkalkbrenner) [reply docker-compose-services/drupal-search_api_solr-cloud/README.md]: outdated

File: docker-compose-services/drupal-search_api_solr_nlp-cloud/docker-compose.drupal-search_api_solr_nlp-cloud.yaml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    # Name of container using standard ddev convention
+    container_name: ddev-${DDEV_SITENAME}-solr
+    # The solr docker image is at https://hub.docker.com/_/solr/
+    # and code at https://github.com/docker-solr/docker-solr
+    # README: https://github.com/docker-solr/docker-solr/blob/master/README.md
+    # It's almost impossible to work with it if you don't read the docs there
+    image: solr:8
+    restart: "no"
+    # Solr is served from this port inside the container.
+    ports:
```
[the remark this answers is not in the document]
Comment (mkalkbrenner) [reply docker-compose-services/drupal-search_api_solr_nlp-cloud/docker-compose.drupal-search_api_solr_nlp-cloud.yaml]: done

File: docker-compose-services/drupal-search_api_solr-cloud/README.md (hunk 1 above)
Comment (mkalkbrenner) [reply docker-compose-services/drupal-search_api_solr-cloud/README.md]: not needed anymore

File: docker-compose-services/drupal-search_api_solr_nlp-cloud/README.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# Search API Solr specific Apache Solr Cloud Integration for DDEV-Local
+
+Although ddev has [documented generic Solr support](https://ddev.readthedocs.io/en/stable/users/extend/additional-services/#apache-solr) it is as simple as possible, and supports only a single core named "dev".
+
+Search API Solr NLP provides a ready to use jump-start config-sets for Solr ion stand-alone and in Cloud mode.
```
[the remark this answers is not in the document]
Comment (mkalkbrenner) [reply docker-compose-services/drupal-search_api_solr_nlp-cloud/README.md]: outdated

**Comment by seifertT** [comment]
I tested the Solr Cloud Integration by @mkalkbrenner and discovered an issue regarding the dynamic IP address allocation for each container where a Solr instance stores the last used IP address. Retested it with the latest commit that uses fixed host names and encountered no more issues - it works like expected now.

**Review by rfay — APPROVED** [review]
Thanks for the great work on this and the review!

**Pull request merged**